trainer/lib/trainer/test_parser.rb

```diff
@@ -202,7 +202,12 @@ def parse_xcresult(path, output_remove_retry_attempts: false)
       path = Shellwords.escape(path)
 
       # Executes xcresulttool to get JSON format of the result bundle object
-      result_bundle_object_raw = execute_cmd("xcrun xcresulttool get --format json --path #{path}")
+      # Hotfix: From Xcode 16 beta 3 'xcresulttool get --format json' has been deprecated; '--legacy' flag required to keep on using the command
+      xcresulttool_cmd = "xcrun xcresulttool get --format json --path #{path}"
+      xcode_version_object = JSON.parse(`plutil -convert json -o - #{FastlaneCore::Helper.xcode_path}../version.plist`)
+      xcresulttool_cmd << ' --legacy' if xcode_version_object['CFBundleVersion'].to_i >= 23044 # CFBundleVersion of Xcode 16 beta 3
```

Comment:
Could potentially change to look at the tool version itself, see https://github.com/davidahouse/XCResultKit/pull/52/files#diff-323dfc344bcc9a79dc779a072e206692565a32efdb3c2c9e2d24c8e80cdd5e8dR125-R134